api_manager.py

```diff
@@ -0,0 +1,28 @@
+import os
+import openai
+import backoff
+
+
+class APIManager:
+    """ A class to manage the API calls to the OpenAI api. """
+
+    def __init__(self):
+        """ initialization.
+        Set the API key from an environment variable.
+        """
+        openai.api_key = os.getenv("OPENAI_API_KEY")  # an environment variable
```

Comment:
It would be better for the environment variable to be in a config file, or at least a constant and not magically appear here